lib/src/component_declaration/function_component.dart

```diff
@@ -74,32 +81,28 @@ export 'component_type_checking.dart'
 ///   (props) {
 ///     return Dom.div()('prop id: ${props.id}');
 ///   },
-///   null,
-///   displayName: 'Foo',
+///   FunctionComponentConfig(
+///     displayName: 'Foo',
+///   ),
 /// );
 /// ```
 ///
 /// Learn more: <https://reactjs.org/docs/components-and-props.html#function-and-class-components>.
 // TODO: right now only top level factory declarations will generate props configs.
 UiFactory<TProps> uiFunction<TProps extends UiProps>(
   dynamic Function(TProps props) functionComponent,
-  FunctionComponentConfig<TProps> config, {
-  PropsFactory<TProps> propsFactory,
-  String displayName,
-}) {
-  if (config != null) {
-    if (propsFactory != null) {
-      throw ArgumentError('propsFactory cannot be used along with config');
-    }
-    propsFactory = config.propsFactory;
-    displayName ??= config.displayName;
+  FunctionComponentConfig<TProps> config) {
+  if (config == null) {
+    throw ArgumentError('config must not be null');
   }
```

Comment:
#nit There's a `.notNull` constructor that should be used
```dart
  if (config == null) {
    throw ArgumentError.notNull('config');
  }
```
There's also a convenience method for it!
```suggestion
  ArgumentError.checkNotNull(config, 'config');
```

Reply:
ooh that's really nice!